_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,91 @@
+
+
+The following strategies/behavioural advices can help to achieve this goal and at the same time improve the team spirit and culture.
+
+
+## Consult - Don't Preach
+
+The basic mindset in which an engineer should approach a code review for a change introduced by another engineer is the mindset of a __consultant__. Whatever you say or write: do it with caution. Sentences that start with _I would approach the problem this way..._ or _I believe this way would be more efficient_ are a good approach if you want to encourage the reviewee to accept your recommendations. Way better than _This solution is inappropriate!_ or _Do it like this!_. Command-style comments are guaranteed to result in damage, conflicts, decreased creativity and motivation and bad team spirit. Don't do it.
+
+By the way: using _please_ to soften a a harsh sentence does not help. You already clearly communicated a difference in status. You already switched your consultant-hat for something that is simply inappropriate in modern software development teams. The result will be a bad taste in the mouth at best.
+
+Even more, given as a command, even if correct, the advice won't be accepted wholeheartedly and the team morale is damaged.
+
+
+If the reviewer would have invested just a little more time and watched the way he communicated: the reviewee might have been very thankful and might have accepted all advise with a smile.
+
+
+The effort is well invested if one takes a break from time to time and makes sure the mindset is tuned, communication style is respectful and reflects the intention to help.
+
+
+## Don't Enforce Consensus
+
+Engineers often tend to forget that there is more than one valid opinion in a lot of cases. __Not everything is determined to false and true__.
+
+So if the reviewer has communicated his opinion appropriate and was not able to convince the reviewee, the code review is the wrong forum to let discussion escalate and enforce consensus. If the code is robust and fulfills the requirements, let it pass the review and try to move the topic out of the situation. You may have tech meetings or other forums in which you can discuss the topic (ideally in general and __not__ by specific example) with a broader audience.
+
+Letting the topic lie for a week can also help. Often times you realize that consensus __isn't important at all__. That not only there are more valid opinions than yours. But also that your team maybe can afford to __let both opinions stand__.
+
+## Don't Point At Rules - Give Explanations
+
+Rules and processes kill __creativity__. You want to give as much freedom to everyone as possible without hurting someone else (that's part of the German Constitution by the way ;)).
+
+Developers develop processes and methods. They align on conventions. And other developers violate those. Often times even conscientiously. How to handle those violations?
```

Comment:
> They align on conventions and other developers violate them; often even

Now perhaps you wanted to say "deliberately", rather than "conscientiously"?